bounceview/src/main/java/com/github/hariprasanths/bounceview/CustomDialog.java

```diff
@@ -0,0 +1,243 @@
+        titleText.setText(text);
+        titleText.setVisibility(Component.VISIBLE);
+        this.setTitleCustomComponent((DirectionalLayout) baseTitleLayout);
+        return this;
+    }
+
+    @Override
+    public CommonDialog setContentText(String text) {
+        Text contentText = (Text) baseContentLayout.findComponentById(ResourceTable.Id_contentText);
+        contentText.setText(text);
+        contentText.setVisibility(0);
+        this.setContentCustomComponent(baseContentLayout);
+        return this;
+    }
+
+    @Override
+    public CommonDialog setContentImage(int resId) {
+        Image image = (Image) baseContentLayout.findComponentById(ResourceTable.Id_imgContent);
+        image.setPixelMap(resId);
+        image.setVisibility(0);
+        this.setContentCustomComponent(baseContentLayout);
+        return this;
+    }
+
+    /**
+     * Does some thing in old style.
+     * use {@link
+     * #setCommonButton(int buttonNum,
+     * String text, int leftMargin, int rightMargin,
+     * Component.ClickedListener listener)} ()}
+     * instead.
+     */
+    @Override
+    public CommonDialog setButton(int buttonNum, String text, ClickedListener listener)
+            throws UnsupportedOperationException {
+        String message = "setButton is unsupported, use setCommonButton(int buttonNum, String text, int leftMargin,"
+                + " int rightMargin, Component.ClickedListener listener) instead.";
+        throw new UnsupportedOperationException(message);
+    }
+
+    /**
+     * set buttons for the custom common dialog.
+     *
+     * @param buttonNum by default at most 3 buttons can be added, takes 0,1,2.
+     * @param text text to display on button
+     * @param leftMargin left margin for button
+     * @param rightMargin right margin for button
+     * @param listener component listener
+     * @return Custom common dialog
+     */
+    public CommonDialog setCommonButton(int buttonNum, String text, int leftMargin, int rightMargin,
+                                        Component.ClickedListener listener) {
+        Button button;
+        switch (buttonNum) {
+            case 0 :
+                button = (Button) baseButtonLayout.findComponentById(ResourceTable.Id_bb0);
+                break;
+            case 1 :
+                button = (Button) baseButtonLayout.findComponentById(ResourceTable.Id_bb1);
+                break;
+            case 2:
+                button = (Button) baseButtonLayout.findComponentById(ResourceTable.Id_bb2);
+                break;
+            default :
+                return this;
+        }
+        button.setVisibility(0);
+        button.setText(text);
+        button.setMarginsLeftAndRight(leftMargin, rightMargin);
+        button.setClickedListener(listener);
+        this.setButtonCustomComponent(baseButtonLayout);
+        return this;
+    }
+
+    /**
+     * Does some thing in old style.
```

Comment:
Make the java doc as "This method is deprecated. Use setCommonImageButton()"

Reply:
done